batchflow/tests/datasetindex_test.py

```diff
@@ -0,0 +1,219 @@
+import sys
+import pytest
+import numpy as np
+
+sys.path.append('..')
+from batchflow.dsindex import DatasetIndex
+
+
+SIZE = 10
+
+@pytest.fixture
+def dsindex_int():
+	return DatasetIndex(SIZE)
+
+@pytest.fixture
+def dsindex_list():
+	return DatasetIndex(list(range(2*SIZE, 4*SIZE)))
+
+@pytest.fixture
+def dsindex_callable():
+	np.random.seed(0)
+	return DatasetIndex(lambda: np.random.random(SIZE))
+
+@pytest.fixture
+def dsindex_big():
+	return DatasetIndex(SIZE**3)
+
+
+
+class TestBaseset:
+
+	def test_baseset_len(self, dsindex_int, dsindex_list, dsindex_callable):
+		dsindex_rec = DatasetIndex(dsindex_list)
+		assert len(dsindex_int) == SIZE
+		assert len(dsindex_list) == 2*SIZE
+		assert len(dsindex_callable) == SIZE
+		assert len(dsindex_rec) == 2*SIZE
+
+	def test_baseset_calc_split_shares(self, dsindex_big):
+		with pytest.raises(ValueError):
+			dsindex_big.calc_split(shares=[0.5, 0.5, 0.5])
+		with pytest.raises(ValueError):
+			dsindex_big.calc_split(shares=[0.5, 0.5, 0.5, 0.5]) 
+		with pytest.raises(ValueError):
+			DatasetIndex(2).calc_split(shares=[0.5, 0.5, 0.5])
+
+	def test_baseset_calc_split_correctness_1(self, dsindex_big):
+		assert sum(dsindex_big.calc_split()) == SIZE**3
+
+	@pytest.mark.skip(reason='default value is 1, not 0. line 94 of batchflow/base.py')
```

Comment:
`xfail` mark is more suitable in this case. `skip` means that you expect your test to pass only if some conditions are met, `xfail` means that test is expected to fail anyway. See this [article](https://docs.pytest.org/en/latest/skipping.html).

Check `skip` marks below too.